artemis/modules/api_scanner.py

```diff
@@ -0,0 +1,168 @@
+    "/swagger.json",
+    "/v2/swagger.json",
+    "/v3/api-docs",
+    "/openapi.json",
+    "/api-docs",
+    "/api/docs" "/docs/swagger.json",
+]
+
+
+class APIResult(BaseModel):
+    url: str
+    method: str
+    vulnerable: bool
+    details: Optional[str]
+    curl_command: Optional[str]
+    status_code: Optional[int]
+
+
+@load_risk_class.load_risk_class(load_risk_class.LoadRiskClass.MEDIUM)
+class APIScanner(ArtemisBase):
+    """
+    Scans API endpoints for vulnerabilities based on OpenAPI/Swagger specifications.
+    """
+
+    identity = "api_scanner"
+    filters = [
+        {"type": TaskType.SERVICE.value, "service": Service.HTTP.value},
+    ]
+
+    def discover_spec(self, base_url: str) -> Optional[str]:
+        """Try to discover OpenAPI/Swagger specification from common paths."""
+        for path in COMMON_SPEC_PATHS:
+            try_url = base_url.rstrip("/") + path
+            try:
+                response = http_requests.get(try_url)
+                if response.status_code == 200 and (
+                    "openapi" in response.text.lower() or "swagger" in response.text.lower()
+                ):
+                    with tempfile.NamedTemporaryFile(delete=False, suffix=".json") as f:
+                        f.write(response.content)
+                        temp_file = f.name
+
+                    # Validate the spec
+                    spec_dict, _ = read_from_filename(temp_file)
+                    try:
+                        validate(spec_dict)
+                        return temp_file
+                    except Exception as e:
+                        self.log.info(f"Unable to validate spec at {try_url}: {e}")
+                        continue
+            except Exception as e:
+                self.log.debug(f"Error checking {try_url}: {e}")
+                continue
+        return None
+
+    def install_offat(self) -> None:
+        if not os.path.exists("/offat"):
+            os.system("git clone https://github.com/OWASP/OFFAT /offat")
+            patch_file_path = os.path.join(os.path.dirname(__file__), "data/offat", "offat_artemis.patch")
+            os.system(f"git -C /offat apply {patch_file_path}")
+
+        os.system("pip install -e /offat/src")
+
+    def scan(self, target_api_specification: str) -> Dict[str, Any]:
+        output_file = "/tmp/output.json"
+
+        if Config.Modules.APIScanner.ONLY_GET_REQUESTS:
+            os.system(
+                " ".join(
+                    [
+                        "offat",
+                        "-f",
+                        target_api_specification,
+                        "--only-get-requests",
+                        "-o",
+                        output_file,
+                        "--format",
+                        "json",
+                    ]
+                )
+            )
+        else:
+            os.system(" ".join(["offat", "-f", target_api_specification, "-o", output_file, "--format", "json"]))
```

Comment:
you can have one subprocess.call invocation and add --only-get-requests to the command, that way we won't make the mistake of updating only one call() ;)